### PR TITLE
Construction menu grid view

### PR DESCRIPTION
**Pull request by qwerltaz**
## About the PR

Add button to construction menu for changing item view to grid. Also resize construction window to be thinner and higher by default, giving view on more items both in normal view and grid view, without sacrificing any part of recipe info on the right.

Good visibility: When selected, the texture of the button goes green for when entity takes up the entire button, and the background goes gray in case of small items.

## Why / Balance

A lot more concise, all items are faster to access. It's added as an optional toggleable feature for those who want to use it.

## Technical details

new case, if grid view button is pressed, item list is cleared and item grid is populated instead. Grid has buttons that call SelectGridButton to handle UI changes.

## Media

https://github.com/user-attachments/assets/30748025-c67e-4905-9669-07ccdb5f6f9f

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

:cl:
- add: A new grid item view is available in the construction menu, togglable with a button.
- tweak: Construction menu default window size was tweaked.

### CODE REVIEW

File: Content.Client/Construction/UI/ConstructionMenuPresenter.cs [hunk 1]
```diff
@@ -201,12 +214,79 @@ private void OnViewPopulateRecipes(object? sender, (string search, string catago
...
+
+            _constructionView.RecipesGridScrollContainer.Visible = _constructionView.GridViewButtonPressed;
+            _constructionView.Recipes.Visible = !_constructionView.GridViewButtonPressed;
+
+            if (_constructionView.GridViewButtonPressed)
             {
-                recipesList.Add(GetItem(recipe, recipesList));
+                foreach (var recipe in recipes)
+                {
+                    var item = GetItem(recipe, recipesList);
```
Comment (chromiumboy) [inline Content.Client/Construction/UI/ConstructionMenuPresenter.cs:230]: There's no need to call this, you can get the icon directly from the recipe

File: Content.Client/Construction/UI/ConstructionMenuPresenter.cs [hunk 2]
```diff
@@ -201,12 +214,79 @@ private void OnViewPopulateRecipes(object? sender, (string search, string catago
...
+
+                    var itemButton = new TextureButton
+                    {
+                        TextureNormal = item.Icon,
```
Comment (chromiumboy) [inline Content.Client/Construction/UI/ConstructionMenuPresenter.cs:234]: See the above

File: Content.Client/Construction/UI/ConstructionMenuPresenter.cs [hunk 3]
```diff
@@ -201,12 +214,79 @@ private void OnViewPopulateRecipes(object? sender, (string search, string catago
...
+                                oldButton.Pressed = false;
+                                SelectGridButton(oldButton, false);
+                                if (_selected == recipe)
+                                {
+                                    OnGridViewRecipeSelected(this, null);
+                                    return;
+                                }
+                            }
+                        }
+                        OnGridViewRecipeSelected(this, buttonToggledEventArgs.Pressed ? recipe : null);
```
Comment (chromiumboy) [inline Content.Client/Construction/UI/ConstructionMenuPresenter.cs:265]: This section is a bit convoluted. Try this:

```
SelectGridButton(itemButton, buttonToggledEventArgs.Pressed);

if (buttonToggledEventArgs.Pressed &&
    _selected != null &&
    _recipeButtons.TryGetValue(_selected.Name, out var oldButton))
{
    oldButton.Pressed = false;
    SelectGridButton(oldButton, false);
}

OnGridViewRecipeSelected(this, buttonToggledEventArgs.Pressed ? recipe : null);
```

**Review by chromiumboy — CHANGES_REQUESTED** [review]

**Comment by chromiumboy** [comment]
I like this a lot, nice work

File: Content.Client/Construction/UI/ConstructionMenuPresenter.cs (hunk 3 above)
Comment (qwerltaz) [reply Content.Client/Construction/UI/ConstructionMenuPresenter.cs]: Nice, but when you exit and re-enter grid view with a button already selected, it forgets it's selected and you can't unselect it. Thankfully, I was able to fix this by adding `itemButton.Pressed = _selected == recipe` in the populating loop. Changes applied.

File: Content.Client/Construction/UI/ConstructionMenuPresenter.cs [hunk 4]
```diff
@@ -223,15 +223,15 @@ private void OnViewPopulateRecipes(object? sender, (string search, string catago
             _constructionView.RecipesGridScrollContainer.Visible = _constructionView.GridViewButtonPressed;
             _constructionView.Recipes.Visible = !_constructionView.GridViewButtonPressed;
 
+            var spriteSystem = _systemManager.GetEntitySystem<SpriteSystem>();
```
Comment (chromiumboy) [inline Content.Client/Construction/UI/ConstructionMenuPresenter.cs:226]: SpriteSystem should be listed as a dependency at the top of the file with the rest of them

**Review by chromiumboy — CHANGES_REQUESTED** [review]
Nice, only one small thing to fix

File: Content.Client/Construction/UI/ConstructionMenuPresenter.cs (hunk 4 above)
Comment (chromiumboy) [reply Content.Client/Construction/UI/ConstructionMenuPresenter.cs:226]: Also noted that on line 380, it's still calling the obsolete Frame0 property on the recipe icon

**Review by chromiumboy — APPROVED** [review]

**Pull request merged**